### ISSUE TITLE
Cached responses 200 -> 304

### ISSUE THREAD

**Issue by n10000k** [nestjs/terminus#2328]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When using a health check route `@HealthCheck()` within a controller the response is cached.

It originally returns the 200 status code then as now the result is cached it's returning 304.

This will cause issues for kubernetes and other services trying to read 200 status code, then maybe retrying to get a cached result for of 304.

### Minimum reproduction code

```ts
  @Get('ping')
  @HealthCheck()
  public ping(): string {
    return 'pong';
  }
```

### Expected behavior

Set `Cache-Control` to `no-store`

### Package version

latest

### NestJS version

latest

### Node.js version

18.1

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

A work around for people is to apply `@Header('Cache-Control', 'no-store')` under `@HealthCheck()`

**Comment by mattgear** [comment]
I had this issue too and had to use the work around.

**Comment by BrunnerLivio** [comment]
After further considerations I have implemented this feature #2335

**Comment by BrunnerLivio** [comment]
Released with [v10.1.0](https://github.com/nestjs/terminus/releases/tag/10.1.0) 🎉
